### PR TITLE
Adding NVIDIA HPC SDK 23.3

### PR DESCRIPTION
**Pull request by fspiga**
Release Notes: https://docs.nvidia.com/hpc-sdk/hpc-sdk-release-notes/index.html

### CODE REVIEW

**Review by samcmill — DISMISSED** [review]

File: var/spack/repos/builtin/packages/nvhpc/package.py [hunk 1]
```diff
@@ -291,8 +305,8 @@ class Nvhpc(Package):
 
     homepage = "https://developer.nvidia.com/hpc-sdk"
 
-    maintainers("samcmill")
-    tags = ["e4s"]
+    maintainers = ["samcmill"]
+    tag = ["e4s"]
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nvhpc/package.py]: Why this change? Can you please revert these 2 lines?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/nvhpc/package.py (hunk 1 above)
Comment (fspiga) [reply var/spack/repos/builtin/packages/nvhpc/package.py]: Done.

File: var/spack/repos/builtin/packages/nvhpc/package.py [hunk 2]
```diff
@@ -292,7 +306,7 @@ class Nvhpc(Package):
     homepage = "https://developer.nvidia.com/hpc-sdk"
 
     maintainers("samcmill")
-    tags = ["e4s"]
+    tags = "e4s"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nvhpc/package.py]:
```suggestion
    tags = ["e4s"]
```

**Review by alalazo — APPROVED** [review]
